From 720cb9faae621011b1094ce52b7fd221325def77 Mon Sep 17 00:00:00 2001
From: Jakub Filak <jfilak@redhat.com>
Date: Tue, 6 May 2014 17:00:04 +0200
Subject: [LIBREPORT PATCH] create augeas lens for libreport

Related to #205

Signed-off-by: Jakub Filak <jfilak@redhat.com>

mmilata: stop configure if augparse not found

Conflicts:

	configure.ac
---
 Makefile.am               |  9 ++++++++-
 augeas/libreport.aug      | 26 ++++++++++++++++++++++++++
 augeas/test_libreport.aug | 47 +++++++++++++++++++++++++++++++++++++++++++++++
 configure.ac              | 14 ++++++++++++++
 4 files changed, 95 insertions(+), 1 deletion(-)
 create mode 100644 augeas/libreport.aug
 create mode 100644 augeas/test_libreport.aug

diff --git a/Makefile.am b/Makefile.am
index 6648eff..332b8a2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,11 +4,18 @@ SUBDIRS = po src tests doc
 EXTRA_DIST = \
     libreport.pc.in \
     libreport-version \
-    asciidoc.conf
+    asciidoc.conf \
+    augeas/test_libreport.aug
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libreport.pc
 
+augeasdir = $(AUGEAS_LENS_LIB_DIR)
+dist_augeas_DATA = augeas/libreport.aug
+
+check-local:
+	$(AUGPARSE) -I $(top_builddir)/augeas $(top_builddir)/augeas/test_libreport.aug
+
 RPM_DIRS = --define "_sourcedir `pwd`" \
 	   --define "_rpmdir `pwd`" \
 	   --define "_specdir `pwd`" \
diff --git a/augeas/libreport.aug b/augeas/libreport.aug
new file mode 100644
index 0000000..dd5ba61
--- /dev/null
+++ b/augeas/libreport.aug
@@ -0,0 +1,26 @@
+module Libreport =
+    autoload xfm
+
+    (* Define useful primitives *)
+    let value_sep    = del / = ?/ " = "
+    let value_to_eol = store /([^ \t\n].*[^ \t\n]|[^ \t\n]?)/
+    let eol          = del /\n/ "\n"
+    let ident        = /[a-zA-Z][a-zA-Z_]+/
+
+    (* Define comment *)
+    let comment = [ label "#comment" . del /#[ \t]*/ "# " . value_to_eol . eol ]
+
+    (* Define empty *)
+    let empty = [ del /[ \t]*\n/ "\n" ]
+
+    (* Define option *)
+    let option = [ key ident . value_sep . value_to_eol . eol ]
+
+    (* Define lens *)
+    let lns = ( comment | empty | option )*
+
+    let filter = (incl "/etc/libreport/plugins/*")
+               . (incl (Sys.getenv("HOME") . "/.config/abrt/settings/*"))
+               . Util.stdexcl
+
+    let xfm = transform lns filter
diff --git a/augeas/test_libreport.aug b/augeas/test_libreport.aug
new file mode 100644
index 0000000..a49474e
--- /dev/null
+++ b/augeas/test_libreport.aug
@@ -0,0 +1,47 @@
+module Test_libreport =
+
+    let conf ="# Bugzilla URL
+BugzillaURL = https://partner-bugzilla.redhat.com/
+# yes means that ssl certificates will be checked
+SSLVerify = yes
+# your login has to exist, if you don have any, please create one
+Login = jfilak@redhat.com
+# your password
+Password =
+
+# SELinux guys almost always move filed bugs from component
+# selinux-policy to another component.
+# This setting instructs reporter-bugzilla to not require
+# component match when it searches for duplicates of
+# bugs in selinux-policy component.
+# (If you need to add more, the syntax is: \"component[,component...]\")
+#
+DontMatchComponents = selinux-policy
+
+# for more info about these settings see: https://github.com/abrt/abrt/wiki/FAQ#creating-private-bugzilla-tickets
+# CreatePrivate = no
+# PrivateGroups = private
+"
+
+    test Libreport.lns get conf =
+        { "#comment" = "Bugzilla URL" }
+        { "BugzillaURL" = "https://partner-bugzilla.redhat.com/" }
+        { "#comment" = "yes means that ssl certificates will be checked" }
+        { "SSLVerify" = "yes" }
+        { "#comment" = "your login has to exist, if you don have any, please create one" }
+        { "Login" = "jfilak@redhat.com" }
+        { "#comment" = "your password" }
+        { "Password" = "" }
+        {}
+        { "#comment" = "SELinux guys almost always move filed bugs from component" }
+        { "#comment" = "selinux-policy to another component." }
+        { "#comment" = "This setting instructs reporter-bugzilla to not require" }
+        { "#comment" = "component match when it searches for duplicates of" }
+        { "#comment" = "bugs in selinux-policy component." }
+        { "#comment" = "(If you need to add more, the syntax is: \"component[,component...]\")" }
+        { "#comment" = "" }
+        { "DontMatchComponents" = "selinux-policy" }
+        {}
+        { "#comment" = "for more info about these settings see: https://github.com/abrt/abrt/wiki/FAQ#creating-private-bugzilla-tickets" }
+        { "#comment" = "CreatePrivate = no" }
+        { "#comment" = "PrivateGroups = private" }
diff --git a/configure.ac b/configure.ac
index 1bd596a..6ef0c5a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -112,6 +112,20 @@ AC_ARG_WITH(debugdumpsdir,
                            [Directory where debugdumps are created])],
             [DEBUG_DUMPS_DIR="$withval"])
 
+AC_ARG_WITH(augeaslenslibdir,
+            [AS_HELP_STRING([--with-augeaslenslibdir=DIR],
+                           [Directory for librepor lens (default: /usr/share/augeas/lenses)])],
+            [], [with_augeaslenslibdir="/usr/share/augeas/lenses"])
+AC_SUBST([AUGEAS_LENS_LIB_DIR], [$with_augeaslenslibdir])
+AC_PATH_PROG(AUGPARSE, augparse, no)
+[if test "$AUGPARSE" = "no"]
+[then]
+    [echo "The augparse program was not found in the search path. Please ensure"]
+    [echo "that it is installed and its directory is included in the search path."]
+    [echo "Then run configure again before attempting to build libreport."]
+    [exit 1]
+[fi]
+
 AC_SUBST(CONF_DIR)
 AC_SUBST(VAR_RUN)
 AC_SUBST(PLUGINS_CONF_DIR)
-- 
1.8.3.1

